utilities/Create_NLI_dataset_from_QA.ipynb

```diff
@@ -0,0 +1,3083 @@
+      "INFO:sentence_transformers.SentenceTransformer:Load pretrained SentenceTransformer: paraphrase-multilingual-mpnet-base-v2\n",
+      "INFO:sentence_transformers.SentenceTransformer:Use pytorch device: cuda\n"
+     ]
+    }
+   ],
+   "source": [
+    "# This function useful for sorting the closest distance\n",
+    "# by using embedding\n",
+    "\n",
+    "model_similarity = SentenceTransformer(MODEL_SIMILARITY_NAME)\n",
+    "\n",
+    "def return_similarity_sorted_array(right_answer, sentence_array, model=model_similarity):\n",
+    "    \n",
+    "    right_answer = right_answer.lower()\n",
+    "    \n",
+    "    embedding_right_answer = model.encode([right_answer], convert_to_tensor=True, device=device)\n",
+    "    embedding_sentence_array = model.encode(sentence_array, convert_to_tensor=True, device=device)\n",
+    "    \n",
+    "    # Using cosine scores to calculate\n",
+    "    cosine_scores = util.pytorch_cos_sim(embedding_right_answer, embedding_sentence_array)\n",
+    "    \n",
+    "    sorted_indices = cosine_scores.argsort(descending=True)[0]\n",
+    "    sorted_array = [sentence_array[i] for i in sorted_indices]\n",
+    "    \n",
+    "    return sorted_array"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "id": "0a9f59dd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# This function useful for\n",
+    "# removing value with hash.\n",
+    "# Because, from label-tagging before\n",
+    "# Some data have a hash symbol, because\n",
+    "# that data was part of a word fragment\n",
+    "\n",
+    "def remove_values_with_hash(arr):\n",
+    "    return [item for item in arr if \"#\" not in item]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "id": "bf6c62a9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Retrieve stopword from all language\n",
+    "\n",
+    "response = requests.get(URL_STOPWORD)\n",
+    "\n",
+    "if response.status_code == 200:\n",
+    "    stopword_data = response.json()\n",
+    "else:\n",
+    "    print(\"Failed to download JSON.\")\n",
+    "\n",
+    "stopword_data = set([item for sublist in list(stopword_data.values()) for item in sublist])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "id": "c043ec1d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# This function just retrieve random word\n",
+    "# of entire premise\n",
+    "\n",
+    "def select_random_word(text, answer, stopword_data=stopword_data):\n",
+    "\n",
+    "    words = re.findall(r'\\w+', text.lower())\n",
+    "    \n",
+    "    # Filtering to remove stopword and punctuation\n",
+    "    filtered_words = [word for word in words if word not in stopword_data and word not in string.punctuation]\n",
+    "    \n",
+    "    # If filtered words less than answer\n",
+    "    # only take one word as random word\n",
+    "    \n",
+    "    if len(filtered_words) < len(answer.split()):\n",
```

Comment:
Ini maksudnya gimana ya? `filtered_words` itu list of words dari premise aja kan? Tujuan dibandingkan dengan answer length apa ya?

Reply:
Karena, pada diskusi Zoom pekan lalu (27/09/2023), Pak Rahmad meminta untuk menyelaraskan bila `right_answer` berbentuk kalimat, maka `wrong_answer` sintesisnya juga berbentuk kalimat, begitu juga pun untuk `right_answer` berbentuk kata, maka `wrong_answer` sintesisnya juga berbentuk kata.

Oleh karena itu, saya pikir untuk, bagaimana cara menyesuaikannya? Salah satu caranya adalah kode tersebut, yang berusaha solving: untuk dapat mengembalikan banyak kata `wrong_answer` yang sama dengan `right_answer`.

Namun, bagian `if` tersebut untuk melakukan pengecekan apakah kata-kata `filtered_words` panjangnya sesuai dengan panjang `right_answer`, kalau panjang `filtered_words` di bawah panjang `right_answer`, maka ambil saja satu kata acak dari `filtered_words`. Kalau masuk ke bagian `else`, maka panjang `wrong_answer` akan disesuaikan dengan `right_answer` dengan mengambil kata dari `filtered_words`.

Reply:
Tapi perbandingannya tetap gak masuk akal. Yang Ravi bandingkan itu kan jumlah kata dalam paragraf dengan jumlah kata dalam jawaban. Misal ada premise "Bandung adalah ibukota Jawa Barat", dengan answer "Bandung". Dengan kode yang sekarang, `filtered_words`-nya `["Bandung", "adalah", "ibukota", "Jawa", "Barat"]`, lalu answer-nya `["Bandung"]`. Padahal kan kita mau memastikan biar panjang kata antara answer dan **random word yang dipilih** itu sama. Kalau memang mau membandingkan length-nya, yg dibandingkan itu satu per satu elemen dari `filtered_words`-nya. Bukan pakai `len(filtered_words)`.

Reply:
Maksud saya, yang saya mau bandingkan itu jumlah kata-nya Kak, bukan length dari suatu kalimat Kak.

Menurut saya, logika saya sudah benar Kak. Karena:

**Suatu kalimat, dapat disebut kalimat jika memiliki >1 kata. Sedangkan suatu kata dapat disebut kata, jika memiliki hanya satu kata.**

Dengan logika itu, maka sudah benar apa yang kita inginkan: **perbandingan jumlah kata**, untuk memisahkan antara kata dan kalimat (banyak kata).

Reply:
Menurut saya Kak, suatu kalimat disebut kalimat karena ada keterkaitan antara kalimat dan kata (banyak kata baru bisa disebut kalimat), bukan keterkaitan antara banyak huruf (atau panjang kata/length of words) Kak..

Misal: 

- Mempertanggungjawabkan
 -> Disini ada 22 huruf (len of words == 22), namun tidak bisa disebut sebagai kalimat karena hanya memiliki satu kata.
- Kucing saya lucu -> Disini ada 14 huruf, namun dapat disebut sebagai kalimat karena memiliki > 1 kata.